### PR TITLE
Refs[TASK][34794]Handle click icon favourite

### PR DESCRIPTION
**Pull request by GiangGiang1105**
## Related Tickets
- [#Task 34794: Handle click icon favourite](https://edu-redmine.sun-asterisk.vn/issues/34794)
## WHAT
- Handle click icon favourite
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/gfood/data/source/MealFavouriteDataSource.kt [hunk 1]
```diff
@@ -7,7 +7,7 @@ interface MealFavouriteDataSource {
 
     interface Local {
         fun save(
```
Comment (phong10101998) [inline app/src/main/java/com/example/gfood/data/source/MealFavouriteDataSource.kt]:
```suggestion
    interface Local {
    
        fun save(
```

File: app/src/main/java/com/example/gfood/screen/mealdetail/MealDetailActivity.kt [hunk 2]
```diff
@@ -23,12 +23,14 @@ class MealDetailActivity : YouTubeBaseActivity(), ContractMealDetail.View {
     private var presenter: MealDetailPresenter? = null
     private var meal: Meal? = null
     private var mealDetail: MealDetail? = null
+    private var statusFavourite: Boolean = false
```
Comment (phong10101998) [inline app/src/main/java/com/example/gfood/screen/mealdetail/MealDetailActivity.kt]:
```suggestion
    private var isStatusFavourite = false
```

File: app/src/main/java/com/example/gfood/data/source/MealFavouriteDataSource.kt (hunk 1 above)
Comment (phong10101998) [reply app/src/main/java/com/example/gfood/data/source/MealFavouriteDataSource.kt]: Thống nhất là cách xuống hay không, không được chỗ có chỗ không có !

File: app/src/main/java/com/example/gfood/data/source/remote/fetchjson/ParseJson.kt [hunk 3]
```diff
@@ -28,7 +28,7 @@ class ParseJson {
         val listIngredient = mutableListOf<String>()
         for (n in 1..10) {
             jsonObject?.getString(MealDetailEntry.INGREDIENT + n)?.let {
-                if (it != null && it.isEmpty()) listIngredient.add(it)
+                if (it != null && !it.isEmpty()) listIngredient.add(it)
```
Comment (phong10101998) [inline app/src/main/java/com/example/gfood/data/source/remote/fetchjson/ParseJson.kt]: use !isNullOrEmpty()

**Review by phong10101998 — APPROVED** [review]

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**